src/main/scala/nl.knaw.dans.easy.dd2d/DepositMigrationTask.scala

```diff
@@ -81,18 +81,18 @@ class DepositMigrationTask(deposit: Deposit,
     trace(persistentId)
     for {
       optAmd <- deposit.tryOptAmd
-      optPublicationDate <- getJsonLdPublicationdate(optAmd)
-      _ <- instance.dataset(persistentId).releaseMigrated(optPublicationDate.get)
+      amd = optAmd.getOrElse(throw new Exception(s"no AMD found for $persistentId"))
```

Comment:
Good idea, but use a more descriptive Exception subclass, in this case `IllegalArgumentException` is fine I think, as we are dealing with erroneous input.